CONTRIBUTING.md

```diff
@@ -71,15 +71,36 @@ git status
 git commit # type a good commit message
 ```
 
-6. Push your commit to GitHub
+6. Bring master up-to-date and rebase
+
+```sh
+git checkout master
+git pull https://github.com/decred/dcrtime.git
+git checkout my-great-stuff
+git rebase -i master
+```
+
+In the text editor, change the command from `pick` to `squash` for **all but the top** commit. If you want to use a different
+commit message for everything, change the command from  `pick` to `reword` on the top commit.
+
+7. **If you have conflicts**, you will need to resolve them by iterating through the diffs one conflicting commit at a time
+
+```sh
+# resolve diffs
+git add file1.go, file2.go, ...
+git rebase --continue
+# repeat until rebase completes
+```
+
+8. Push your commit to GitHub
 
 Assuming `origin` is the name of the remote used for *your* repository:

```

Comment:
I wonder if we should change `origin` to `myremote` because by no means certain `origin` is the user's fork.  In fact, my workflow has github.com/decred/dcrdata as `origin`, while I have my fork as the `chappjc` remote (same for lots of other forks of contributors).